auth/events.go

```diff
@@ -0,0 +1,35 @@
+// Copyright © 2018 Banzai Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package auth
+
+import "github.com/banzaicloud/pipeline/internal/events"
+
+const OrganizationRegisteredTopic = "organization_registered"
+
+// authEvents is responsible for dispatching domain events throughout the system.
+// It does not express any infrastructural detail (like pubsub).
+type authEventPublisher interface {
```

Comment:
authEvents seems to be a better name from a domain point of view. You don't need to care about different event responsibilities here other than publishing.

Either way, the interface name should match the name in the comment.